src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -1737,14 +1738,35 @@ public void setThisThread(Thread thisThread) {
     this.thisThread = thisThread;
   }
 
-  public ServiceInterface startPeer(String reservedKey) {
-    String actualName = getPeerName(reservedKey);
-    return Runtime.start(actualName, null);
+  public ServiceInterface startPeer(String peerKey) {
+    String actualName = getPeerName(peerKey);
+    ServiceInterface si = Runtime.start(actualName, null);
+    if (si != null) {
+      ServiceReservation sr = serviceType.getPeer(peerKey);
+      if (sr != null) {
+        sr.state = "STARTED";
+        sr.actualName = actualName;
+        sr.type = si.getSimpleName();
+        if (si != null) {
+          CodecUtils.setField(this, peerKey, si);
```

Comment:
why does this need to do fancy introspection ?  I'm missing something that is causing this additional level of complexity.

Reply:
good question :)
It establishes a convention and contract ... 
If a service has a field named the same as a peer 
it will be set when the parent is started ...

Already, the convention/contract was established that peers start before their parents.
there is a bunch of npe's and various manual conventions that were established which tried to assign the reference of a peer to a class member.

This takes the random developers implementation out of the mix.

Rule is:
* if you went through the effort to define a peer for a service, and you have a local reference named the same thing
e.g. transient SerialDevice serial ... then when the parent is started, the framework assigns the reference

Reply:
fancy...